backend/kernelCI_app/views/treeCommitsHistory.py

```diff
@@ -1,13 +1,110 @@
 from rest_framework.views import APIView
 from rest_framework.response import Response
 from rest_framework import status
-from django.http import JsonResponse
+from django.http import JsonResponse, HttpResponseBadRequest
 from django.db import connection
+from kernelCI_app.utils import (
+    FilterParams,
+    InvalidComparisonOP,
+    getErrorResponseBody,
+)
 
 
 # TODO Move this endpoint to a function so it doesn't
 # have to be another request, it can be called from the tree details endpoint
 class TreeCommitsHistory(APIView):
+    def __init__(self):
+        self.filters_options = {
+            'build': {
+                'table_alias': 'b',
+                'filters': []
+            },
+            'boot': {
+                'table_alias': 't',
+                'filters': []
+            },
+            'test': {
+                'table_alias': 't',
+                'filters': []
+            }
+        }
+        self.field_values = dict()
+
+    def get_filters(self, filter_params):
+        for f in filter_params.filters:
```

Comment:
this logic seems very similar to the one that is being used already, can you add a TODO to unite the filters logic:?